platform/linux-generic/odp_ipsec.c

```diff
@@ -1354,6 +1443,17 @@ static int ipsec_out_esp(odp_packet_t *pkt,
 				 esptrl_offset, _ODP_ESPTRL_LEN,
 				 &esptrl);
 
+	/* Outbound ICV computation includes ESN higher 32 bits as part of ESP
+	 * implicit trailer for individual algo's.
+	 */
+	if (ipsec_sa->esn && !ipsec_sa->combined_mode) {
+		uint32_t outb_seqh = (seq_no >> 32);
+
+		odp_packet_copy_from_mem(*pkt,
+					 esptrl_offset + _ODP_ESPTRL_LEN,
+					 IPSEC_SEQ_HI_LEN, &outb_seqh);
```

Comment:
It seems this may attempt to copy beyond the end of the packet if ICV length is less than 4 (0 for null authentication). Checkin the insert_seq_hi flag as I suggested would help a bit. But this is buggy otherwise too, see below.

Reply:
Yes, checking with "insert_seq_hi" will be correct which is zero for "ODP_AUTH_ALG_NULL" case( before, this code block used to check with "seqh_len" which was zero for "ODP_AUTH_ALG_NULL"), I feel this should be sufficient here.